src/pkg/cli/tail.go

```diff
@@ -265,6 +279,12 @@ func Tail(ctx context.Context, client client.Client, service, etag string, since
 					line = pkg.StripAnsi(line)
 				}
 				fmt.Println(line)
+
+				// Detect end logging event
+				if params.EndEventDetectFunc != nil && params.EndEventDetectFunc(msg.Service, msg.Host, line) {
+					cancel()
+					return nil
```

Comment:
End tail function on detection of event log matching what we are looking for (text to show deployment/teardown succeeded).

Reply:
actually that `cancel` triggered a Golang lint error, which suggested to always defer: https://github.com/defang-io/defang/pull/374